src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -19412,6 +19412,361 @@ public void AbstractInSubmission()
+            
+            var codeC2 = @"
+.assembly C
+{
+	.ver 0:0:0:0
+}
+.assembly extern D1 { }
+.assembly extern D2 { }
+.class extern forwarder C.ClassC
+{
+	.assembly extern D1
+}
+.class extern forwarder C.ClassC
+{
+	.assembly extern D2
+}";
+
+            var referenceC2 = CompileIL(codeC2, appendDefaultHeader: false);
+
+            CreateCompilationWithMscorlib(codeA, references: new MetadataReference[] { referenceB, referenceC2 }, assemblyName: "A").VerifyDiagnostics(
+                // (10,13): error CS8205: Type 'C.ClassC' is forwarded to multiple assemblies: 'D1' and 'D2'
+                //             ClassB.MethodB(null);
+                Diagnostic(ErrorCode.ERR_TypeForwardedToMultipleAssemblies, "ClassB.MethodB").WithArguments("C.ClassC", "D1", "D2").WithLocation(10, 13));
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleTypeForwardersToTheSameAssemblyShouldNotResultInMultipleForwardError()
+        {
+            var codeC = @"
+namespace C
+{
+    public class ClassC {}
+}";
+            var referenceC = CreateCompilationWithMscorlib(codeC, assemblyName: "C").EmitToImageReference();
+
+            var codeB = @"
+using C;
+
+namespace B
+{
+    public static class ClassB
+    {
+        public static string MethodB(ClassC obj)
+        {
+            return ""obj is "" + (obj == null ? ""null"" : obj.ToString());
+        }
+    }
+}";
+            var referenceB = CreateCompilationWithMscorlib(codeB, references: new MetadataReference[] { referenceC }, assemblyName: "B").EmitToImageReference();
+
+            var codeA = @"
+using B;
+
+namespace A
+{
+    public class ClassA
+    {
+        public static void Main()
+        {
+            System.Console.WriteLine(ClassB.MethodB(null));
+        }
+    }
+}";
+
+            CompileAndVerify(
+                source: codeA,
+                additionalRefs: new MetadataReference[] { referenceB, referenceC },
+                expectedOutput: "obj is null");
+
+            var codeC2 = @"
+.assembly C
+{
+	.ver 0:0:0:0
+}
+.assembly extern D { }
+.class extern forwarder C.ClassC
+{
+	.assembly extern D
+}
+.class extern forwarder C.ClassC
+{
+	.assembly extern D
+}";
+
+            var referenceC2 = CompileIL(codeC2, appendDefaultHeader: false);
+
+            CreateCompilationWithMscorlib(codeA, references: new MetadataReference[] { referenceB, referenceC2 }).VerifyDiagnostics(
+                // (10,38): error CS0012: The type 'ClassC' is defined in an assembly that is not referenced. You must add a reference to assembly 'D, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
+                //             System.Console.WriteLine(ClassB.MethodB(null));
+                Diagnostic(ErrorCode.ERR_NoTypeDef, "ClassB.MethodB").WithArguments("C.ClassC", "D, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(10, 38));
```

Comment:
>Diagnostic(ErrorCode.ERR_NoTypeDef, "ClassB.MethodB").WithArguments("C.ClassC", "D, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(10, 38)); [](start = 16, length = 162)

Now let's try adding a reference to an assembly D, we should be able to compile and execute.